### ISSUE TITLE
nginx ingress controller fails to come up on 2.2-head

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#23196]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
Deploy an rke AWS cluster - 3 etcd, 1 control plane and 3 worker nodes.

**Expected Result:**
When the cluster comes up Active, the Ingress controller should also be active under Cluster --> System project --> Inginx ngress controller

**Actual Result:**
The Ingress controller fails to come up. -- Crashloop
Ingress controller logs:
```
...............
E1002 22:39:39.823870       6 reflector.go:125] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:179: Failed to list *v1beta1.Ingress: ingresses.networking.k8s.io is forbidden: User "system:serviceaccount:ingress-nginx:nginx-ingress-serviceaccount" cannot list resource "ingresses" in API group "networking.k8s.io" at the cluster scope
E1002 22:39:40.825279       6 reflector.go:125] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:179: Failed to list *v1beta1.Ingress: ingresses.networking.k8s.io is forbidden: User "system:serviceaccount:ingress-nginx:nginx-ingress-serviceaccount" cannot list resource "ingresses" in API group "networking.k8s.io" at the cluster scope
E1002 22:39:41.659161       6 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I1002 22:39:41.677596       6 main.go:154] Received SIGTERM, shutting down
I1002 22:39:41.677620       6 nginx.go:401] Shutting down controller queues
I1002 22:39:41.677634       6 status.go:117] updating status of Ingress rules (remove)
E1002 22:39:41.677837       6 store.go:183] timed out waiting for caches to sync
I1002 22:39:41.677872       6 nginx.go:318] Starting NGINX process
I1002 22:39:41.678215       6 leaderelection.go:235] attempting to acquire leader lease  ingress-nginx/ingress-controller-leader-nginx...
E1002 22:39:41.678538       6 queue.go:78] queue has been shutdown, failed to enqueue: &ObjectMeta{Name:initial-sync,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,ManagedFields:[],}
I1002 22:39:41.688571       6 leaderelection.go:245] successfully acquired lease ingress-nginx/ingress-controller-leader-nginx
I1002 22:39:41.688724       6 status.go:86] new leader elected: nginx-ingress-controller-4prrw
E1002 22:39:41.688768       6 queue.go:78] queue has been shutdown, failed to enqueue: &ObjectMeta{Name:sync status,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,ManagedFields:[],}
I1002 22:39:41.702751       6 status.go:127] leaving status update for next leader (3)
I1002 22:39:41.702765       6 nginx.go:417] Stopping NGINX process
2019/10/02 22:39:41 [notice] 31#31: signal process started
2019/10/02 22:39:41 [error] 31#31: open() "/tmp/nginx.pid" failed (2: No such file or directory)
nginx: [error] open() "/tmp/nginx.pid" failed (2: No such file or directory)
I1002 22:39:41.707506       6 main.go:158] Error during shutdown: exit status 1
I1002 22:39:41.707519       6 main.go:162] Handled quit, awaiting Pod deletion
E1002 22:39:42.843907       6 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I1002 22:39:51.707652       6 main.go:165] Exiting with 1
```

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2-head - commit id: **`1375ef26`**
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): rke AWS
- Kubernetes version (use `kubectl version`):

```
1.15.4
```

- Docker version (use `docker version`):

```
18.9.9
```

**Comment by alena1108** [comment]
@prachidamle we need to update nginx ingress template

**Comment by prachidamle** [comment]
yes, but not all changes from the nginx ingress template on master can be applied to v2.2 since master has changes due to k8s1.16 and some windows related too.

**Comment by sangeethah** [comment]
This issue is seen with all K8s versions - `v1.14.7`, `v1.13.11` and `v1.15.4`

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.2-head 9aea30e5

As a standard user, I add three EC2 clusters which have the following k8s versions
```
- 1.15.4 - default 
- 1.14.7
- 1.13.11
```
and each cluster has 
```
- 1 etcd
- 1 control plane
- 3 worker nodes. 
```

Results for all three clusters:
- the cluster is active
- the Ingress controller is active
- no errors in the Ingress controller's log
- no restarts for pods in the Ingress controller workload
